py/desiutil/install.py

```diff
@@ -909,6 +909,28 @@ def get_extra(self):
                     raise DesiInstallException(message)
         return
 
+    def compile_branch(self):
+        """Certain packages need C/C++ code compiled even for a branch install.
+        """
+        if self.is_branch:
+            compile_script = os.path.join(self.install_dir, 'etc',
+                                          '{0}_compile.sh'.format(self.baseproduct))
+            if os.path.exists(compile_script):
+                self.log.debug("Detected compile script: %s.", compile_script)
+                if self.options.test:
+                    self.log.debug('Test Mode. Skipping compile script.')
+                else:
+                    proc = Popen([compile_script, sys.executable], universal_newlines=True,
```

Comment:
The example script in desihub/fiberassign#435 includes `cd ${INSTALL_DIR}` so that it moves to the top-level directory before running `python setup.py build_ext --inplace`.  Although that works, it introduces a coupling between package scripts needing to know about environment variables temporarily set by `desiInstall` but not usually set in any other context.  How about handling that within desiInstall itself, e.g.
```python
current_dir = os.getcwd()
os.chdir(self.install_dir)
proc = Popen(...)
os.chdir(current_dir)
```
i.e. we would guarantee that the script is run from the top-level directory and the script itself doesn't have to figure out how to get there (but it could do additional relative `cd` if it needed, e.g. to descend into a subdir for some other compile/config step).

Reply:
I'm not _opposed_ to this suggestion, but it would create a policy inconsistency, because the _data_ install scripts _are_ allowed to know about `${INSTALL _DIR}`. Just look at the desimodel script.

Reply:
Thanks for pointing out the $INSTALL_DIR usage in desimodel/etc/desimodel_data.sh.  At least in that case the use of $INSTALL_DIR is optional (if it isn't set it just skips that step).

I'm not proposing to go change the data scripts usage here, but at minimum let's not further propagate an $INSTALL_DIR coupling into new scripts in additional repos.